framework/src/vectorpostprocessors/SideValueSampler.C

```diff
@@ -53,12 +74,38 @@ SideValueSampler::initialize()
 void
 SideValueSampler::execute()
 {
-  for (unsigned int _qp = 0; _qp < _qrule->n_points(); _qp++)
+  if (_qp_sampling)
+    for (unsigned int _qp = 0; _qp < _qrule->n_points(); _qp++)
+    {
+      for (unsigned int i = 0; i < _coupled_moose_vars.size(); i++)
+        _values[i] = (dynamic_cast<MooseVariable *>(_coupled_moose_vars[i]))->sln()[_qp];
+
+      SamplerBase::addSample(_q_point[_qp], _current_elem->id(), _values);
+    }
+  else
   {
-    for (unsigned int i = 0; i < _coupled_moose_vars.size(); i++)
-      _values[i] = (dynamic_cast<MooseVariable *>(_coupled_moose_vars[i]))->sln()[_qp];
+    getFaceInfos();
+
+    const auto state = determineState();
+
+    for (const auto & fi : _face_infos)
+    {
+      for (unsigned int i = 0; i < _fv_vars.size(); i++)
+      {
+        mooseAssert(_coupled_moose_vars[i]->hasFaceSide(*fi, true) ||
+                        _coupled_moose_vars[i]->hasFaceSide(*fi, false),
+                    "Variable " + _coupled_moose_vars[i]->name() +
+                        " should be defined on one side of the face!");
+
+        auto * elem = _fv_vars[i]->hasFaceSide(*fi, true) ? fi->elemPtr() : fi->neighborPtr();
+
+        const auto face_arg = Moose::FaceArg(
+            {fi, Moose::FV::LimiterType::CentralDifference, true, false, elem, nullptr});
+        _values[i] = MetaPhysicL::raw_value((*_fv_vars[i])(face_arg, state));
+      }
 
-    SamplerBase::addSample(_q_point[_qp], _current_elem->id(), _values);
+      SamplerBase::addSample(fi->faceCentroid(), fi->id(), _values);
```

Comment:
I d rather we used current_elem -> id() still
fi->id() does not match to anything in paraview

Reply:
Sounds good!